### ISSUE TITLE
Idea: Help catching forgotten strings

### ISSUE THREAD

**Issue by machour** [lingui/js-lingui#186]
Hi @tricoder42 and cheers for this **excellent** library 🙌 

What would you think about implementing some kind of debug mode in js-lingui that can be turned on/off by the developer to quickly spot the places where he forgot to use `<Trans>` ?

Here's what it would look like:

| Normal mode | Debug mode |
| ---------------- | --------------- |
|![simulator screen shot - iphone x - 2018-03-16 at 19 53 02](https://user-images.githubusercontent.com/304450/37539643-8d189b7a-2954-11e8-9d37-c900aa3cbfd4.png) | ![simulator screen shot - iphone x - 2018-03-16 at 19 56 40](https://user-images.githubusercontent.com/304450/37539654-905dcada-2954-11e8-8c7c-923fafc2e63e.png)|

This helps to see that the last string "commented on pull request" was totally forgotten while translating the application.

The number/length of xxxx would match the original string to keep the UI balance.

**Comment by machour** [comment]
Replacing the output of components nested inside `<Trans>` with Xs would make that even nicer 😍 
 
![screen shot 2018-03-16 at 8 12 08 pm](https://user-images.githubusercontent.com/304450/37540206-6a4af0dc-2956-11e8-855d-27b1f552ae60.png)

**Comment by tricoder42** [comment]
Interesting idea 👍 

I'll play with it for a while to see what's the best API.

**Comment by HriBB** [comment]
What about adding an additional command `lingui missing`, which would show missing translation ids.

**Comment by tricoder42** [comment]
My proposal here:

Add a new prop `missing` to `I18nProvider.` This value will be used when translation isn't available, but only in development.

It may be a simple string:

```js
<I18nProvider missing="xxxxxxx"><App /></I18nProvider>
```

or a fancy component which reports all missing translations (pseudo code):

```js
const missing = {}

function DebugToolbarSink({ language, id }) {
   missing[language] = missing[language] || {}
   missing[language][id] = true
   return '⚠️'
}

function DebugToolbar() {
  return <ul>{Object.keys(missing[language]).map(id => <li>id</id>)}</ul>
}

<I18nProvider missing={DebugToolbarSink}>
   <App />
   <DebugToolbar />
</I18nProvider>
```

**Comment by tricoder42** [comment]
@HriBB That's also an interesting idea. We could leverage info about origin of messages and either show compact version (missing IDs) or verbose one (missing ID and location in code)

**Comment by tricoder42** [comment]
Released in [2.4.0](https://github.com/lingui/js-lingui/releases/tag/v2.4.0)
